10/03/2018 22:43:28.0792	[raft_python_callback.py write_to_logger 70]  792 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:init_state_functions_handler, LINE:28
MESSAGE:start

10/03/2018 22:43:28.0792	[raft_python_callback.py write_to_logger 70]  792 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:init_state_functions_handler, LINE:70
MESSAGE:end

10/03/2018 22:43:28.0792	[raft_python_callback.py write_to_logger 70]  792 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:init_multicast_message, LINE:81
MESSAGE:init multicast socket

10/03/2018 22:43:28.0793	[raft_python_callback.py write_to_logger 70]  793 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:init_multicast_message, LINE:88
MESSAGE:multicast was opened successfully

10/03/2018 22:43:28.0793	[raft_python_callback.py write_to_logger 70]  793 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:init_raft_queue, LINE:13
MESSAGE:first time

10/03/2018 22:43:28.0794	[raft_python_callback.py write_to_logger 70]  794 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:16
MESSAGE:update DB
VARIABLES:  DB_flag=status  key=status 

10/03/2018 22:43:28.0794	[raft_python_callback.py write_to_logger 70]  794 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:17
MESSAGE:update DB
VARIABLES:  value=-3 

10/03/2018 22:43:28.0796	[raft_python_callback.py write_to_logger 70]  796 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:init_raft, LINE:190
MESSAGE:init succeed

10/03/2018 22:43:28.0796	[raft_python_callback.py write_to_logger 70]  796 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_raft, LINE:231
MESSAGE:start raft...
VARIABLES:  raft_ip=230.1.1.10 

10/03/2018 22:43:28.0796	[raft_python_callback.py write_to_logger 70]  796 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_raft, LINE:233
MESSAGE:start raft...
VARIABLES:  raft_port=12345  server_id=2  members_num=2  leader_timeout=1000 

10/03/2018 22:43:28.0796	[raft_python_callback.py write_to_logger 70]  796 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_raft, LINE:239
MESSAGE:run raft...

10/03/2018 22:43:28.0796	[raft_python_callback.py write_to_logger 70]  796 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:43:31.0231	[raft_python_callback.py write_to_logger 70]  231 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=0 

10/03/2018 22:43:31.0231	[raft_python_callback.py write_to_logger 70]  231 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=0 

10/03/2018 22:43:33.0664	[raft_python_callback.py write_to_logger 70]  664 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=0 

10/03/2018 22:43:33.0664	[raft_python_callback.py write_to_logger 70]  664 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=0 

10/03/2018 22:43:33.0664	[raft_python_callback.py write_to_logger 70]  664 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:16
MESSAGE:update DB
VARIABLES:  DB_flag=status  key=term 

10/03/2018 22:43:33.0664	[raft_python_callback.py write_to_logger 70]  664 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:17
MESSAGE:update DB
VARIABLES:  value=0 

10/03/2018 22:43:33.0665	[raft_python_callback.py write_to_logger 70]  665 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:16
MESSAGE:update DB
VARIABLES:  DB_flag=status  key=status 

10/03/2018 22:43:33.0665	[raft_python_callback.py write_to_logger 70]  665 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:17
MESSAGE:update DB
VARIABLES:  value=-2 

10/03/2018 22:43:33.0666	[raft_python_callback.py write_to_logger 70]  666 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:16
MESSAGE:update DB
VARIABLES:  DB_flag=status  key=leader_id 

10/03/2018 22:43:33.0666	[raft_python_callback.py write_to_logger 70]  666 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:17
MESSAGE:update DB
VARIABLES:  value=0 

10/03/2018 22:43:33.0667	[raft_python_callback.py write_to_logger 70]  667 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=8 

10/03/2018 22:43:33.0667	[raft_python_callback.py write_to_logger 70]  667 root [DEBUG] 
FILE:./C/Code/follower_handlers.c, FUNCTION:follower_time_out_handler, LINE:253
MESSAGE:follower became candidate and is sending request for vote msg

10/03/2018 22:43:33.0667	[raft_python_callback.py write_to_logger 70]  667 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=20 

10/03/2018 22:43:33.0667	[raft_python_callback.py write_to_logger 70]  667 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=20  fixed_message_size=20 

10/03/2018 22:43:33.0667	[raft_python_callback.py write_to_logger 70]  667 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:init_raft_queue, LINE:23
MESSAGE:not first time

10/03/2018 22:43:33.0667	[raft_python_callback.py write_to_logger 70]  667 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=20  fixed_message_size=20 

10/03/2018 22:43:33.0668	[raft_python_callback.py write_to_logger 70]  668 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:43:33.0668	[raft_python_callback.py write_to_logger 70]  668 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:43:33.0884	[raft_python_callback.py write_to_logger 70]  884 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=20  fixed_message_size=20 

10/03/2018 22:43:33.0884	[raft_python_callback.py write_to_logger 70]  884 root [INFO ] 
FILE:./C/Code/handlers_utils.c, FUNCTION:check_if_is_for_me, LINE:103
MESSAGE:got new message for me

10/03/2018 22:43:33.0884	[raft_python_callback.py write_to_logger 70]  884 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:is_relevant_for_candidate, LINE:77
MESSAGE:candidate check if msg is relevant
VARIABLES:  sharedRaftData.raft_state.term=0  sharedRaftData.raft_state.server_id=2  node_message->message_sent_to=2  node_message->term=0  node_message->event=7 

10/03/2018 22:43:33.0884	[raft_python_callback.py write_to_logger 70]  884 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:is_relevant_for_candidate, LINE:91
MESSAGE:candidate:relevant msg

10/03/2018 22:43:33.0884	[raft_python_callback.py write_to_logger 70]  884 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:is_relevant_message, LINE:154
MESSAGE:check if msg is relevant
VARIABLES:  sharedRaftData.raft_state.current_state=1  node_message->event=7  is_relevant_rv=1  node_message->term=0 

10/03/2018 22:43:33.0884	[raft_python_callback.py write_to_logger 70]  884 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=1  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:43:33.0884	[raft_python_callback.py write_to_logger 70]  884 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=7  sharedRaftData.raft_state.current_state=1 

10/03/2018 22:43:33.0884	[raft_python_callback.py write_to_logger 70]  884 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:init_raft_queue, LINE:23
MESSAGE:not first time

10/03/2018 22:43:33.0884	[raft_python_callback.py write_to_logger 70]  884 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:43:33.0885	[raft_python_callback.py write_to_logger 70]  885 root [DEBUG] 
FILE:./C/Code/candidate_handlers.c, FUNCTION:candidate_vote_for_me_handler, LINE:20
MESSAGE:candidate got majority- becoming leader
VARIABLES:  node->event=7 
	 sharedRaftData.raft_state.term=0 
	 sharedRaftData.raft_state.current_state=2 
	 sharedRaftData.raft_state.vote_counter=2 
	 node->message_sent_by=1 
	 node->term=0 
	

10/03/2018 22:43:33.0885	[raft_python_callback.py write_to_logger 70]  885 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:16
MESSAGE:update DB
VARIABLES:  DB_flag=status  key=status 

10/03/2018 22:43:33.0885	[raft_python_callback.py write_to_logger 70]  885 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:17
MESSAGE:update DB
VARIABLES:  value=-1 

10/03/2018 22:43:33.0886	[raft_python_callback.py write_to_logger 70]  886 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:16
MESSAGE:update DB
VARIABLES:  DB_flag=status  key=leader_id 

10/03/2018 22:43:33.0886	[raft_python_callback.py write_to_logger 70]  886 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:17
MESSAGE:update DB
VARIABLES:  value=2 

10/03/2018 22:43:33.0886	[raft_python_callback.py write_to_logger 70]  886 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:43:33.0886	[raft_python_callback.py write_to_logger 70]  886 root [DEBUG] 
FILE:./C/Code/candidate_handlers.c, FUNCTION:candidate_vote_for_me_handler, LINE:40
MESSAGE:candidate sending keep alive hb msg

10/03/2018 22:43:33.0886	[raft_python_callback.py write_to_logger 70]  886 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:43:33.0887	[raft_python_callback.py write_to_logger 70]  887 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:33.0887	[raft_python_callback.py write_to_logger 70]  887 root [DEBUG] 
FILE:./C/Code/candidate_handlers.c, FUNCTION:candidate_vote_for_me_handler, LINE:68
MESSAGE:end of vote event
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:33.0887	[raft_python_callback.py write_to_logger 70]  887 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:33.0887	[raft_python_callback.py write_to_logger 70]  887 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:43:33.0887	[raft_python_callback.py write_to_logger 70]  887 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:43:36.0097	[raft_python_callback.py write_to_logger 70]  97 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:36.0098	[raft_python_callback.py write_to_logger 70]  98 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:36.0098	[raft_python_callback.py write_to_logger 70]  98 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:43:36.0098	[raft_python_callback.py write_to_logger 70]  98 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:43:36.0098	[raft_python_callback.py write_to_logger 70]  98 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:43:36.0098	[raft_python_callback.py write_to_logger 70]  98 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:43:36.0098	[raft_python_callback.py write_to_logger 70]  98 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:36.0099	[raft_python_callback.py write_to_logger 70]  99 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:36.0099	[raft_python_callback.py write_to_logger 70]  99 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:43:36.0099	[raft_python_callback.py write_to_logger 70]  99 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:43:37.0099	[raft_python_callback.py write_to_logger 70]  99 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:37.0100	[raft_python_callback.py write_to_logger 70]  100 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:37.0100	[raft_python_callback.py write_to_logger 70]  100 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:43:37.0100	[raft_python_callback.py write_to_logger 70]  100 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:43:37.0100	[raft_python_callback.py write_to_logger 70]  100 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:43:37.0100	[raft_python_callback.py write_to_logger 70]  100 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:43:37.0101	[raft_python_callback.py write_to_logger 70]  101 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:37.0101	[raft_python_callback.py write_to_logger 70]  101 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:37.0101	[raft_python_callback.py write_to_logger 70]  101 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:43:37.0101	[raft_python_callback.py write_to_logger 70]  101 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:43:38.0101	[raft_python_callback.py write_to_logger 70]  101 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:38.0101	[raft_python_callback.py write_to_logger 70]  101 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:38.0101	[raft_python_callback.py write_to_logger 70]  101 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:43:38.0102	[raft_python_callback.py write_to_logger 70]  102 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:43:38.0102	[raft_python_callback.py write_to_logger 70]  102 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:43:38.0102	[raft_python_callback.py write_to_logger 70]  102 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:43:38.0102	[raft_python_callback.py write_to_logger 70]  102 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:38.0102	[raft_python_callback.py write_to_logger 70]  102 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:38.0102	[raft_python_callback.py write_to_logger 70]  102 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:43:38.0102	[raft_python_callback.py write_to_logger 70]  102 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:43:39.0102	[raft_python_callback.py write_to_logger 70]  102 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:39.0102	[raft_python_callback.py write_to_logger 70]  102 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:39.0102	[raft_python_callback.py write_to_logger 70]  102 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:43:39.0102	[raft_python_callback.py write_to_logger 70]  102 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:43:39.0102	[raft_python_callback.py write_to_logger 70]  102 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:43:39.0103	[raft_python_callback.py write_to_logger 70]  103 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:43:39.0103	[raft_python_callback.py write_to_logger 70]  103 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:39.0103	[raft_python_callback.py write_to_logger 70]  103 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:39.0103	[raft_python_callback.py write_to_logger 70]  103 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:43:39.0103	[raft_python_callback.py write_to_logger 70]  103 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:43:40.0103	[raft_python_callback.py write_to_logger 70]  103 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:40.0103	[raft_python_callback.py write_to_logger 70]  103 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:40.0103	[raft_python_callback.py write_to_logger 70]  103 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:43:40.0103	[raft_python_callback.py write_to_logger 70]  103 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:43:40.0103	[raft_python_callback.py write_to_logger 70]  103 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:43:40.0104	[raft_python_callback.py write_to_logger 70]  104 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:43:40.0104	[raft_python_callback.py write_to_logger 70]  104 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:40.0104	[raft_python_callback.py write_to_logger 70]  104 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:40.0104	[raft_python_callback.py write_to_logger 70]  104 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:43:40.0104	[raft_python_callback.py write_to_logger 70]  104 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:43:41.0105	[raft_python_callback.py write_to_logger 70]  105 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:41.0105	[raft_python_callback.py write_to_logger 70]  105 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:41.0105	[raft_python_callback.py write_to_logger 70]  105 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:43:41.0105	[raft_python_callback.py write_to_logger 70]  105 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:43:41.0105	[raft_python_callback.py write_to_logger 70]  105 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:43:41.0105	[raft_python_callback.py write_to_logger 70]  105 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:43:41.0106	[raft_python_callback.py write_to_logger 70]  106 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:41.0106	[raft_python_callback.py write_to_logger 70]  106 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:41.0106	[raft_python_callback.py write_to_logger 70]  106 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:43:41.0106	[raft_python_callback.py write_to_logger 70]  106 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:43:42.0106	[raft_python_callback.py write_to_logger 70]  106 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:42.0107	[raft_python_callback.py write_to_logger 70]  107 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:42.0107	[raft_python_callback.py write_to_logger 70]  107 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:43:42.0107	[raft_python_callback.py write_to_logger 70]  107 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:43:42.0107	[raft_python_callback.py write_to_logger 70]  107 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:43:42.0107	[raft_python_callback.py write_to_logger 70]  107 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:43:42.0107	[raft_python_callback.py write_to_logger 70]  107 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:42.0107	[raft_python_callback.py write_to_logger 70]  107 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:42.0108	[raft_python_callback.py write_to_logger 70]  108 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:43:42.0108	[raft_python_callback.py write_to_logger 70]  108 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:43:43.0135	[raft_python_callback.py write_to_logger 70]  135 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:43.0135	[raft_python_callback.py write_to_logger 70]  135 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:43.0135	[raft_python_callback.py write_to_logger 70]  135 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:43:43.0135	[raft_python_callback.py write_to_logger 70]  135 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:43:43.0135	[raft_python_callback.py write_to_logger 70]  135 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:43:43.0135	[raft_python_callback.py write_to_logger 70]  135 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:43:43.0135	[raft_python_callback.py write_to_logger 70]  135 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:43.0135	[raft_python_callback.py write_to_logger 70]  135 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:43.0136	[raft_python_callback.py write_to_logger 70]  136 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:43:43.0136	[raft_python_callback.py write_to_logger 70]  136 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:43:44.0138	[raft_python_callback.py write_to_logger 70]  138 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:44.0138	[raft_python_callback.py write_to_logger 70]  138 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:44.0138	[raft_python_callback.py write_to_logger 70]  138 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:43:44.0139	[raft_python_callback.py write_to_logger 70]  139 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:43:44.0139	[raft_python_callback.py write_to_logger 70]  139 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:43:44.0139	[raft_python_callback.py write_to_logger 70]  139 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:43:44.0139	[raft_python_callback.py write_to_logger 70]  139 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:44.0139	[raft_python_callback.py write_to_logger 70]  139 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:44.0139	[raft_python_callback.py write_to_logger 70]  139 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:43:44.0140	[raft_python_callback.py write_to_logger 70]  140 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:43:45.0138	[raft_python_callback.py write_to_logger 70]  138 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:45.0139	[raft_python_callback.py write_to_logger 70]  139 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:45.0139	[raft_python_callback.py write_to_logger 70]  139 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:43:45.0139	[raft_python_callback.py write_to_logger 70]  139 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:43:45.0139	[raft_python_callback.py write_to_logger 70]  139 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:43:45.0139	[raft_python_callback.py write_to_logger 70]  139 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:43:45.0139	[raft_python_callback.py write_to_logger 70]  139 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:45.0139	[raft_python_callback.py write_to_logger 70]  139 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:45.0139	[raft_python_callback.py write_to_logger 70]  139 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:43:45.0139	[raft_python_callback.py write_to_logger 70]  139 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:43:46.0140	[raft_python_callback.py write_to_logger 70]  140 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:46.0140	[raft_python_callback.py write_to_logger 70]  140 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:46.0140	[raft_python_callback.py write_to_logger 70]  140 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:43:46.0140	[raft_python_callback.py write_to_logger 70]  140 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:43:46.0140	[raft_python_callback.py write_to_logger 70]  140 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:43:46.0140	[raft_python_callback.py write_to_logger 70]  140 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:43:46.0141	[raft_python_callback.py write_to_logger 70]  141 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:46.0141	[raft_python_callback.py write_to_logger 70]  141 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:46.0141	[raft_python_callback.py write_to_logger 70]  141 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:43:46.0141	[raft_python_callback.py write_to_logger 70]  141 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:43:47.0141	[raft_python_callback.py write_to_logger 70]  141 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:47.0142	[raft_python_callback.py write_to_logger 70]  142 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:47.0142	[raft_python_callback.py write_to_logger 70]  142 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:43:47.0142	[raft_python_callback.py write_to_logger 70]  142 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:43:47.0142	[raft_python_callback.py write_to_logger 70]  142 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:43:47.0142	[raft_python_callback.py write_to_logger 70]  142 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:43:47.0142	[raft_python_callback.py write_to_logger 70]  142 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:47.0142	[raft_python_callback.py write_to_logger 70]  142 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:47.0142	[raft_python_callback.py write_to_logger 70]  142 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:43:47.0142	[raft_python_callback.py write_to_logger 70]  142 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:43:48.0143	[raft_python_callback.py write_to_logger 70]  143 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:48.0144	[raft_python_callback.py write_to_logger 70]  144 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:48.0144	[raft_python_callback.py write_to_logger 70]  144 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:43:48.0144	[raft_python_callback.py write_to_logger 70]  144 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:43:48.0144	[raft_python_callback.py write_to_logger 70]  144 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:43:48.0144	[raft_python_callback.py write_to_logger 70]  144 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:43:48.0145	[raft_python_callback.py write_to_logger 70]  145 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:48.0145	[raft_python_callback.py write_to_logger 70]  145 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:48.0145	[raft_python_callback.py write_to_logger 70]  145 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:43:48.0145	[raft_python_callback.py write_to_logger 70]  145 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:43:48.0526	[raft_python_callback.py write_to_logger 70]  526 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:start_commit_process, LINE:209
MESSAGE: start commit process

10/03/2018 22:43:48.0527	[raft_python_callback.py write_to_logger 70]  527 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=9  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:48.0527	[raft_python_callback.py write_to_logger 70]  527 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_send_log_hb_handler, LINE:190
MESSAGE:leader send new log entry
VARIABLES:  sharedRaftData.raft_state.last_commit_index=-1  sharedRaftData.raft_state.last_log_index=-1  sharedRaftData.raft_state.commit_counter=0  new_log_commit_id=0 

10/03/2018 22:43:48.0527	[raft_python_callback.py write_to_logger 70]  527 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_send_log_hb_handler, LINE:193
MESSAGE:leader send sync log msg
VARIABLES:  node->msg_data.set_log_hb_msg.cmd=add 
	 node->msg_data.set_log_hb_msg.key=x 
	 node->msg_data.set_log_hb_msg.value=1 
	

10/03/2018 22:43:48.0529	[raft_python_callback.py write_to_logger 70]  529 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:16
MESSAGE:update DB
VARIABLES:  DB_flag=status  key=applied_last_idx 

10/03/2018 22:43:48.0529	[raft_python_callback.py write_to_logger 70]  529 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:17
MESSAGE:update DB
VARIABLES:  value=0 

10/03/2018 22:43:48.0532	[raft_python_callback.py write_to_logger 70]  532 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=2 

10/03/2018 22:43:48.0532	[raft_python_callback.py write_to_logger 70]  532 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_send_log_hb_handler, LINE:207
MESSAGE:leader sending set_log_hb msg

10/03/2018 22:43:48.0532	[raft_python_callback.py write_to_logger 70]  532 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=180 

10/03/2018 22:43:48.0532	[raft_python_callback.py write_to_logger 70]  532 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=180  fixed_message_size=180 

10/03/2018 22:43:48.0532	[raft_python_callback.py write_to_logger 70]  532 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=180  fixed_message_size=180 

10/03/2018 22:43:48.0532	[raft_python_callback.py write_to_logger 70]  532 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:43:48.0532	[raft_python_callback.py write_to_logger 70]  532 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:43:48.0736	[raft_python_callback.py write_to_logger 70]  736 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:48.0737	[raft_python_callback.py write_to_logger 70]  737 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:is_relevant_message, LINE:154
MESSAGE:check if msg is relevant
VARIABLES:  sharedRaftData.raft_state.current_state=2  node_message->event=3  is_relevant_rv=1  node_message->term=0 

10/03/2018 22:43:48.0737	[raft_python_callback.py write_to_logger 70]  737 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=1  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:43:48.0737	[raft_python_callback.py write_to_logger 70]  737 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:43:48.0737	[raft_python_callback.py write_to_logger 70]  737 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=3  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:48.0737	[raft_python_callback.py write_to_logger 70]  737 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_log_res_handler, LINE:221
MESSAGE:leader log response msg
VARIABLES:  sharedRaftData.raft_state.commit_counter=1 
	 sharedRaftData.raft_state.members_amount=2 
	 sharedRaftData.raft_state.last_commit_index=-1 
	

10/03/2018 22:43:48.0738	[raft_python_callback.py write_to_logger 70]  738 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=6 

10/03/2018 22:43:48.0738	[raft_python_callback.py write_to_logger 70]  738 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_log_res_handler, LINE:231
MESSAGE:leader sending commit ok msg

10/03/2018 22:43:48.0738	[raft_python_callback.py write_to_logger 70]  738 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:43:48.0738	[raft_python_callback.py write_to_logger 70]  738 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:48.0738	[raft_python_callback.py write_to_logger 70]  738 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:16
MESSAGE:update DB
VARIABLES:  DB_flag=status  key=commit_idx 

10/03/2018 22:43:48.0738	[raft_python_callback.py write_to_logger 70]  738 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:17
MESSAGE:update DB
VARIABLES:  value=0 

10/03/2018 22:43:48.0739	[raft_python_callback.py write_to_logger 70]  739 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:48.0739	[raft_python_callback.py write_to_logger 70]  739 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:43:48.0739	[raft_python_callback.py write_to_logger 70]  739 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:43:48.0742	[raft_python_callback.py write_to_logger 70]  742 root [CRITI] 
T
10/03/2018 22:43:49.0144	[raft_python_callback.py write_to_logger 70]  144 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:49.0144	[raft_python_callback.py write_to_logger 70]  144 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:49.0144	[raft_python_callback.py write_to_logger 70]  144 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:43:49.0145	[raft_python_callback.py write_to_logger 70]  145 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:43:49.0145	[raft_python_callback.py write_to_logger 70]  145 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:43:49.0145	[raft_python_callback.py write_to_logger 70]  145 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:43:49.0145	[raft_python_callback.py write_to_logger 70]  145 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:49.0145	[raft_python_callback.py write_to_logger 70]  145 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:49.0145	[raft_python_callback.py write_to_logger 70]  145 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:43:49.0145	[raft_python_callback.py write_to_logger 70]  145 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:43:50.0145	[raft_python_callback.py write_to_logger 70]  145 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:50.0146	[raft_python_callback.py write_to_logger 70]  146 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:50.0146	[raft_python_callback.py write_to_logger 70]  146 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:43:50.0146	[raft_python_callback.py write_to_logger 70]  146 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:43:50.0146	[raft_python_callback.py write_to_logger 70]  146 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:43:50.0146	[raft_python_callback.py write_to_logger 70]  146 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:43:50.0146	[raft_python_callback.py write_to_logger 70]  146 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:50.0146	[raft_python_callback.py write_to_logger 70]  146 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:50.0147	[raft_python_callback.py write_to_logger 70]  147 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:43:50.0147	[raft_python_callback.py write_to_logger 70]  147 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:43:51.0146	[raft_python_callback.py write_to_logger 70]  146 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:51.0146	[raft_python_callback.py write_to_logger 70]  146 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:51.0146	[raft_python_callback.py write_to_logger 70]  146 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:43:51.0147	[raft_python_callback.py write_to_logger 70]  147 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:43:51.0147	[raft_python_callback.py write_to_logger 70]  147 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:43:51.0147	[raft_python_callback.py write_to_logger 70]  147 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:43:51.0147	[raft_python_callback.py write_to_logger 70]  147 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:51.0147	[raft_python_callback.py write_to_logger 70]  147 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:51.0147	[raft_python_callback.py write_to_logger 70]  147 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:43:51.0148	[raft_python_callback.py write_to_logger 70]  148 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:43:52.0147	[raft_python_callback.py write_to_logger 70]  147 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:52.0147	[raft_python_callback.py write_to_logger 70]  147 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:52.0147	[raft_python_callback.py write_to_logger 70]  147 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:43:52.0147	[raft_python_callback.py write_to_logger 70]  147 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:43:52.0147	[raft_python_callback.py write_to_logger 70]  147 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:43:52.0147	[raft_python_callback.py write_to_logger 70]  147 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:43:52.0148	[raft_python_callback.py write_to_logger 70]  148 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:52.0148	[raft_python_callback.py write_to_logger 70]  148 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:52.0148	[raft_python_callback.py write_to_logger 70]  148 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:43:52.0148	[raft_python_callback.py write_to_logger 70]  148 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:43:53.0148	[raft_python_callback.py write_to_logger 70]  148 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:53.0148	[raft_python_callback.py write_to_logger 70]  148 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:53.0148	[raft_python_callback.py write_to_logger 70]  148 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:43:53.0148	[raft_python_callback.py write_to_logger 70]  148 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:43:53.0149	[raft_python_callback.py write_to_logger 70]  149 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:43:53.0149	[raft_python_callback.py write_to_logger 70]  149 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:43:53.0149	[raft_python_callback.py write_to_logger 70]  149 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:53.0149	[raft_python_callback.py write_to_logger 70]  149 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:53.0149	[raft_python_callback.py write_to_logger 70]  149 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:43:53.0149	[raft_python_callback.py write_to_logger 70]  149 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:43:54.0150	[raft_python_callback.py write_to_logger 70]  150 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:54.0150	[raft_python_callback.py write_to_logger 70]  150 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:54.0151	[raft_python_callback.py write_to_logger 70]  151 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:43:54.0151	[raft_python_callback.py write_to_logger 70]  151 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:43:54.0151	[raft_python_callback.py write_to_logger 70]  151 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:43:54.0151	[raft_python_callback.py write_to_logger 70]  151 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:43:54.0152	[raft_python_callback.py write_to_logger 70]  152 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:54.0152	[raft_python_callback.py write_to_logger 70]  152 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:54.0152	[raft_python_callback.py write_to_logger 70]  152 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:43:54.0152	[raft_python_callback.py write_to_logger 70]  152 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:43:55.0152	[raft_python_callback.py write_to_logger 70]  152 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:55.0152	[raft_python_callback.py write_to_logger 70]  152 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:55.0152	[raft_python_callback.py write_to_logger 70]  152 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:43:55.0152	[raft_python_callback.py write_to_logger 70]  152 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:43:55.0152	[raft_python_callback.py write_to_logger 70]  152 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:43:55.0153	[raft_python_callback.py write_to_logger 70]  153 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:43:55.0153	[raft_python_callback.py write_to_logger 70]  153 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:55.0153	[raft_python_callback.py write_to_logger 70]  153 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:55.0153	[raft_python_callback.py write_to_logger 70]  153 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:43:55.0153	[raft_python_callback.py write_to_logger 70]  153 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:43:56.0152	[raft_python_callback.py write_to_logger 70]  152 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:56.0153	[raft_python_callback.py write_to_logger 70]  153 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:56.0153	[raft_python_callback.py write_to_logger 70]  153 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:43:56.0153	[raft_python_callback.py write_to_logger 70]  153 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:43:56.0153	[raft_python_callback.py write_to_logger 70]  153 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:43:56.0153	[raft_python_callback.py write_to_logger 70]  153 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:43:56.0153	[raft_python_callback.py write_to_logger 70]  153 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:56.0153	[raft_python_callback.py write_to_logger 70]  153 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:56.0153	[raft_python_callback.py write_to_logger 70]  153 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:43:56.0153	[raft_python_callback.py write_to_logger 70]  153 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:43:57.0153	[raft_python_callback.py write_to_logger 70]  153 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:57.0153	[raft_python_callback.py write_to_logger 70]  153 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:57.0153	[raft_python_callback.py write_to_logger 70]  153 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:43:57.0153	[raft_python_callback.py write_to_logger 70]  153 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:43:57.0153	[raft_python_callback.py write_to_logger 70]  153 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:43:57.0153	[raft_python_callback.py write_to_logger 70]  153 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:43:57.0154	[raft_python_callback.py write_to_logger 70]  154 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:57.0154	[raft_python_callback.py write_to_logger 70]  154 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:57.0154	[raft_python_callback.py write_to_logger 70]  154 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:43:57.0154	[raft_python_callback.py write_to_logger 70]  154 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:43:57.0863	[raft_python_callback.py write_to_logger 70]  863 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:start_commit_process, LINE:209
MESSAGE: start commit process

10/03/2018 22:43:57.0863	[raft_python_callback.py write_to_logger 70]  863 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=9  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:57.0863	[raft_python_callback.py write_to_logger 70]  863 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_send_log_hb_handler, LINE:190
MESSAGE:leader send new log entry
VARIABLES:  sharedRaftData.raft_state.last_commit_index=0  sharedRaftData.raft_state.last_log_index=0  sharedRaftData.raft_state.commit_counter=2  new_log_commit_id=1 

10/03/2018 22:43:57.0863	[raft_python_callback.py write_to_logger 70]  863 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_send_log_hb_handler, LINE:193
MESSAGE:leader send sync log msg
VARIABLES:  node->msg_data.set_log_hb_msg.cmd=add 
	 node->msg_data.set_log_hb_msg.key=xx 
	 node->msg_data.set_log_hb_msg.value=1 
	

10/03/2018 22:43:57.0864	[raft_python_callback.py write_to_logger 70]  864 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:16
MESSAGE:update DB
VARIABLES:  DB_flag=status  key=applied_last_idx 

10/03/2018 22:43:57.0865	[raft_python_callback.py write_to_logger 70]  865 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:17
MESSAGE:update DB
VARIABLES:  value=1 

10/03/2018 22:43:57.0866	[raft_python_callback.py write_to_logger 70]  866 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=2 

10/03/2018 22:43:57.0866	[raft_python_callback.py write_to_logger 70]  866 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_send_log_hb_handler, LINE:207
MESSAGE:leader sending set_log_hb msg

10/03/2018 22:43:57.0866	[raft_python_callback.py write_to_logger 70]  866 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=180 

10/03/2018 22:43:57.0866	[raft_python_callback.py write_to_logger 70]  866 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=180  fixed_message_size=180 

10/03/2018 22:43:57.0866	[raft_python_callback.py write_to_logger 70]  866 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=180  fixed_message_size=180 

10/03/2018 22:43:57.0866	[raft_python_callback.py write_to_logger 70]  866 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:43:57.0867	[raft_python_callback.py write_to_logger 70]  867 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:43:58.0052	[raft_python_callback.py write_to_logger 70]  52 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:58.0052	[raft_python_callback.py write_to_logger 70]  52 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:is_relevant_message, LINE:154
MESSAGE:check if msg is relevant
VARIABLES:  sharedRaftData.raft_state.current_state=2  node_message->event=3  is_relevant_rv=1  node_message->term=0 

10/03/2018 22:43:58.0052	[raft_python_callback.py write_to_logger 70]  52 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=1  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:43:58.0052	[raft_python_callback.py write_to_logger 70]  52 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:43:58.0053	[raft_python_callback.py write_to_logger 70]  53 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=3  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:58.0053	[raft_python_callback.py write_to_logger 70]  53 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_log_res_handler, LINE:221
MESSAGE:leader log response msg
VARIABLES:  sharedRaftData.raft_state.commit_counter=1 
	 sharedRaftData.raft_state.members_amount=2 
	 sharedRaftData.raft_state.last_commit_index=0 
	

10/03/2018 22:43:58.0053	[raft_python_callback.py write_to_logger 70]  53 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=6 

10/03/2018 22:43:58.0053	[raft_python_callback.py write_to_logger 70]  53 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_log_res_handler, LINE:231
MESSAGE:leader sending commit ok msg

10/03/2018 22:43:58.0053	[raft_python_callback.py write_to_logger 70]  53 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:43:58.0053	[raft_python_callback.py write_to_logger 70]  53 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:58.0053	[raft_python_callback.py write_to_logger 70]  53 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:16
MESSAGE:update DB
VARIABLES:  DB_flag=status  key=commit_idx 

10/03/2018 22:43:58.0054	[raft_python_callback.py write_to_logger 70]  54 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:17
MESSAGE:update DB
VARIABLES:  value=1 

10/03/2018 22:43:58.0054	[raft_python_callback.py write_to_logger 70]  54 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:58.0054	[raft_python_callback.py write_to_logger 70]  54 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:43:58.0054	[raft_python_callback.py write_to_logger 70]  54 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:43:58.0058	[raft_python_callback.py write_to_logger 70]  58 root [CRITI] 
T
10/03/2018 22:43:58.0155	[raft_python_callback.py write_to_logger 70]  155 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:58.0155	[raft_python_callback.py write_to_logger 70]  155 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:58.0155	[raft_python_callback.py write_to_logger 70]  155 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:43:58.0155	[raft_python_callback.py write_to_logger 70]  155 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:43:58.0155	[raft_python_callback.py write_to_logger 70]  155 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:43:58.0155	[raft_python_callback.py write_to_logger 70]  155 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:43:58.0155	[raft_python_callback.py write_to_logger 70]  155 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:58.0155	[raft_python_callback.py write_to_logger 70]  155 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:58.0155	[raft_python_callback.py write_to_logger 70]  155 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:43:58.0156	[raft_python_callback.py write_to_logger 70]  156 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:43:59.0157	[raft_python_callback.py write_to_logger 70]  157 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:59.0157	[raft_python_callback.py write_to_logger 70]  157 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:43:59.0157	[raft_python_callback.py write_to_logger 70]  157 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:43:59.0157	[raft_python_callback.py write_to_logger 70]  157 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:43:59.0157	[raft_python_callback.py write_to_logger 70]  157 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:43:59.0158	[raft_python_callback.py write_to_logger 70]  158 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:43:59.0158	[raft_python_callback.py write_to_logger 70]  158 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:59.0158	[raft_python_callback.py write_to_logger 70]  158 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:43:59.0158	[raft_python_callback.py write_to_logger 70]  158 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:43:59.0158	[raft_python_callback.py write_to_logger 70]  158 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:44:00.0165	[raft_python_callback.py write_to_logger 70]  165 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:00.0166	[raft_python_callback.py write_to_logger 70]  166 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:00.0166	[raft_python_callback.py write_to_logger 70]  166 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:44:00.0166	[raft_python_callback.py write_to_logger 70]  166 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:44:00.0166	[raft_python_callback.py write_to_logger 70]  166 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:44:00.0166	[raft_python_callback.py write_to_logger 70]  166 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:44:00.0166	[raft_python_callback.py write_to_logger 70]  166 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:00.0166	[raft_python_callback.py write_to_logger 70]  166 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:00.0166	[raft_python_callback.py write_to_logger 70]  166 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:44:00.0167	[raft_python_callback.py write_to_logger 70]  167 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:44:01.0168	[raft_python_callback.py write_to_logger 70]  168 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:01.0168	[raft_python_callback.py write_to_logger 70]  168 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:01.0168	[raft_python_callback.py write_to_logger 70]  168 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:44:01.0168	[raft_python_callback.py write_to_logger 70]  168 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:44:01.0168	[raft_python_callback.py write_to_logger 70]  168 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:44:01.0168	[raft_python_callback.py write_to_logger 70]  168 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:44:01.0169	[raft_python_callback.py write_to_logger 70]  169 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:01.0169	[raft_python_callback.py write_to_logger 70]  169 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:01.0169	[raft_python_callback.py write_to_logger 70]  169 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:44:01.0169	[raft_python_callback.py write_to_logger 70]  169 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:44:02.0169	[raft_python_callback.py write_to_logger 70]  169 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:02.0169	[raft_python_callback.py write_to_logger 70]  169 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:02.0169	[raft_python_callback.py write_to_logger 70]  169 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:44:02.0169	[raft_python_callback.py write_to_logger 70]  169 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:44:02.0169	[raft_python_callback.py write_to_logger 70]  169 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:44:02.0169	[raft_python_callback.py write_to_logger 70]  169 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:44:02.0169	[raft_python_callback.py write_to_logger 70]  169 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:02.0169	[raft_python_callback.py write_to_logger 70]  169 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:02.0169	[raft_python_callback.py write_to_logger 70]  169 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:44:02.0169	[raft_python_callback.py write_to_logger 70]  169 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:44:03.0170	[raft_python_callback.py write_to_logger 70]  170 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:03.0170	[raft_python_callback.py write_to_logger 70]  170 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:03.0170	[raft_python_callback.py write_to_logger 70]  170 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:44:03.0170	[raft_python_callback.py write_to_logger 70]  170 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:44:03.0171	[raft_python_callback.py write_to_logger 70]  171 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:44:03.0171	[raft_python_callback.py write_to_logger 70]  171 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:44:03.0171	[raft_python_callback.py write_to_logger 70]  171 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:03.0171	[raft_python_callback.py write_to_logger 70]  171 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:03.0171	[raft_python_callback.py write_to_logger 70]  171 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:44:03.0171	[raft_python_callback.py write_to_logger 70]  171 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:44:04.0171	[raft_python_callback.py write_to_logger 70]  171 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:04.0171	[raft_python_callback.py write_to_logger 70]  171 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:04.0172	[raft_python_callback.py write_to_logger 70]  172 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:44:04.0172	[raft_python_callback.py write_to_logger 70]  172 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:44:04.0172	[raft_python_callback.py write_to_logger 70]  172 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:44:04.0172	[raft_python_callback.py write_to_logger 70]  172 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:44:04.0173	[raft_python_callback.py write_to_logger 70]  173 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:04.0173	[raft_python_callback.py write_to_logger 70]  173 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:04.0173	[raft_python_callback.py write_to_logger 70]  173 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:44:04.0173	[raft_python_callback.py write_to_logger 70]  173 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:44:05.0172	[raft_python_callback.py write_to_logger 70]  172 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:05.0172	[raft_python_callback.py write_to_logger 70]  172 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:05.0172	[raft_python_callback.py write_to_logger 70]  172 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:44:05.0172	[raft_python_callback.py write_to_logger 70]  172 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:44:05.0172	[raft_python_callback.py write_to_logger 70]  172 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:44:05.0172	[raft_python_callback.py write_to_logger 70]  172 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:44:05.0172	[raft_python_callback.py write_to_logger 70]  172 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:05.0172	[raft_python_callback.py write_to_logger 70]  172 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:05.0173	[raft_python_callback.py write_to_logger 70]  173 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:44:05.0173	[raft_python_callback.py write_to_logger 70]  173 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:44:06.0173	[raft_python_callback.py write_to_logger 70]  173 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:06.0173	[raft_python_callback.py write_to_logger 70]  173 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:06.0173	[raft_python_callback.py write_to_logger 70]  173 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:44:06.0173	[raft_python_callback.py write_to_logger 70]  173 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:44:06.0174	[raft_python_callback.py write_to_logger 70]  174 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:44:06.0174	[raft_python_callback.py write_to_logger 70]  174 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:44:06.0174	[raft_python_callback.py write_to_logger 70]  174 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:06.0174	[raft_python_callback.py write_to_logger 70]  174 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:06.0174	[raft_python_callback.py write_to_logger 70]  174 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:44:06.0174	[raft_python_callback.py write_to_logger 70]  174 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:44:07.0174	[raft_python_callback.py write_to_logger 70]  174 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:07.0175	[raft_python_callback.py write_to_logger 70]  175 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:07.0175	[raft_python_callback.py write_to_logger 70]  175 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:44:07.0175	[raft_python_callback.py write_to_logger 70]  175 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:44:07.0175	[raft_python_callback.py write_to_logger 70]  175 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:44:07.0175	[raft_python_callback.py write_to_logger 70]  175 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:44:07.0176	[raft_python_callback.py write_to_logger 70]  176 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:07.0176	[raft_python_callback.py write_to_logger 70]  176 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:07.0176	[raft_python_callback.py write_to_logger 70]  176 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:44:07.0176	[raft_python_callback.py write_to_logger 70]  176 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:44:08.0177	[raft_python_callback.py write_to_logger 70]  177 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:08.0177	[raft_python_callback.py write_to_logger 70]  177 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:08.0177	[raft_python_callback.py write_to_logger 70]  177 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:44:08.0177	[raft_python_callback.py write_to_logger 70]  177 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:44:08.0178	[raft_python_callback.py write_to_logger 70]  178 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:44:08.0178	[raft_python_callback.py write_to_logger 70]  178 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:44:08.0178	[raft_python_callback.py write_to_logger 70]  178 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:08.0178	[raft_python_callback.py write_to_logger 70]  178 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:08.0178	[raft_python_callback.py write_to_logger 70]  178 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:44:08.0178	[raft_python_callback.py write_to_logger 70]  178 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:44:09.0178	[raft_python_callback.py write_to_logger 70]  178 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:09.0178	[raft_python_callback.py write_to_logger 70]  178 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:09.0179	[raft_python_callback.py write_to_logger 70]  179 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:44:09.0179	[raft_python_callback.py write_to_logger 70]  179 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:44:09.0179	[raft_python_callback.py write_to_logger 70]  179 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:44:09.0179	[raft_python_callback.py write_to_logger 70]  179 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:44:09.0180	[raft_python_callback.py write_to_logger 70]  180 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:09.0180	[raft_python_callback.py write_to_logger 70]  180 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:09.0180	[raft_python_callback.py write_to_logger 70]  180 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:44:09.0180	[raft_python_callback.py write_to_logger 70]  180 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:44:10.0179	[raft_python_callback.py write_to_logger 70]  179 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:10.0180	[raft_python_callback.py write_to_logger 70]  180 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:10.0180	[raft_python_callback.py write_to_logger 70]  180 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:44:10.0180	[raft_python_callback.py write_to_logger 70]  180 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:44:10.0180	[raft_python_callback.py write_to_logger 70]  180 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:44:10.0180	[raft_python_callback.py write_to_logger 70]  180 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:44:10.0181	[raft_python_callback.py write_to_logger 70]  181 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:10.0181	[raft_python_callback.py write_to_logger 70]  181 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:10.0181	[raft_python_callback.py write_to_logger 70]  181 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:44:10.0181	[raft_python_callback.py write_to_logger 70]  181 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:44:11.0180	[raft_python_callback.py write_to_logger 70]  180 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:11.0181	[raft_python_callback.py write_to_logger 70]  181 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:11.0181	[raft_python_callback.py write_to_logger 70]  181 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:44:11.0181	[raft_python_callback.py write_to_logger 70]  181 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:44:11.0181	[raft_python_callback.py write_to_logger 70]  181 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:44:11.0181	[raft_python_callback.py write_to_logger 70]  181 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:44:11.0181	[raft_python_callback.py write_to_logger 70]  181 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:11.0181	[raft_python_callback.py write_to_logger 70]  181 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:11.0182	[raft_python_callback.py write_to_logger 70]  182 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:44:11.0182	[raft_python_callback.py write_to_logger 70]  182 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:44:12.0181	[raft_python_callback.py write_to_logger 70]  181 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:12.0181	[raft_python_callback.py write_to_logger 70]  181 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:12.0181	[raft_python_callback.py write_to_logger 70]  181 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:44:12.0181	[raft_python_callback.py write_to_logger 70]  181 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:44:12.0182	[raft_python_callback.py write_to_logger 70]  182 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:44:12.0182	[raft_python_callback.py write_to_logger 70]  182 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:44:12.0182	[raft_python_callback.py write_to_logger 70]  182 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:12.0182	[raft_python_callback.py write_to_logger 70]  182 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:12.0182	[raft_python_callback.py write_to_logger 70]  182 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:44:12.0182	[raft_python_callback.py write_to_logger 70]  182 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:44:13.0182	[raft_python_callback.py write_to_logger 70]  182 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:13.0182	[raft_python_callback.py write_to_logger 70]  182 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:13.0182	[raft_python_callback.py write_to_logger 70]  182 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:44:13.0182	[raft_python_callback.py write_to_logger 70]  182 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:44:13.0182	[raft_python_callback.py write_to_logger 70]  182 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:44:13.0182	[raft_python_callback.py write_to_logger 70]  182 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:44:13.0182	[raft_python_callback.py write_to_logger 70]  182 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:13.0183	[raft_python_callback.py write_to_logger 70]  183 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:13.0183	[raft_python_callback.py write_to_logger 70]  183 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:44:13.0183	[raft_python_callback.py write_to_logger 70]  183 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:44:14.0182	[raft_python_callback.py write_to_logger 70]  182 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:14.0182	[raft_python_callback.py write_to_logger 70]  182 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:14.0183	[raft_python_callback.py write_to_logger 70]  183 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:44:14.0183	[raft_python_callback.py write_to_logger 70]  183 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:44:14.0183	[raft_python_callback.py write_to_logger 70]  183 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:44:14.0183	[raft_python_callback.py write_to_logger 70]  183 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:44:14.0183	[raft_python_callback.py write_to_logger 70]  183 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:14.0183	[raft_python_callback.py write_to_logger 70]  183 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:14.0183	[raft_python_callback.py write_to_logger 70]  183 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:44:14.0183	[raft_python_callback.py write_to_logger 70]  183 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:44:15.0183	[raft_python_callback.py write_to_logger 70]  183 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:15.0183	[raft_python_callback.py write_to_logger 70]  183 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:15.0183	[raft_python_callback.py write_to_logger 70]  183 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:44:15.0184	[raft_python_callback.py write_to_logger 70]  184 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:44:15.0184	[raft_python_callback.py write_to_logger 70]  184 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:44:15.0184	[raft_python_callback.py write_to_logger 70]  184 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:44:15.0184	[raft_python_callback.py write_to_logger 70]  184 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:15.0184	[raft_python_callback.py write_to_logger 70]  184 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:15.0184	[raft_python_callback.py write_to_logger 70]  184 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:44:15.0185	[raft_python_callback.py write_to_logger 70]  185 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:44:15.0353	[raft_python_callback.py write_to_logger 70]  353 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=28  fixed_message_size=28 

10/03/2018 22:44:15.0353	[raft_python_callback.py write_to_logger 70]  353 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:is_relevant_message, LINE:154
MESSAGE:check if msg is relevant
VARIABLES:  sharedRaftData.raft_state.current_state=2  node_message->event=4  is_relevant_rv=1  node_message->term=0 

10/03/2018 22:44:15.0354	[raft_python_callback.py write_to_logger 70]  354 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=1  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:44:15.0354	[raft_python_callback.py write_to_logger 70]  354 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:44:15.0354	[raft_python_callback.py write_to_logger 70]  354 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=4  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:15.0354	[raft_python_callback.py write_to_logger 70]  354 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_sync_req_handler, LINE:12
MESSAGE:leader get sync request msg
VARIABLES:  sharedRaftData.raft_state.last_commit_index=1  sharedRaftData.raft_state.last_log_index=1  node->msg_data.sync_req_msg.start_log_index=0  node->msg_data.sync_req_msg.last_log_id=1  node->message_sent_by=1 

10/03/2018 22:44:15.0355	[raft_python_callback.py write_to_logger 70]  355 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_sync_req_handler, LINE:37
MESSAGE:first elemnt - command
VARIABLES:  elem=add 

10/03/2018 22:44:15.0356	[raft_python_callback.py write_to_logger 70]  356 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_sync_req_handler, LINE:45
MESSAGE:message pointer and element key/value
VARIABLES:  msg_pointer=add  elem=x 

10/03/2018 22:44:15.0356	[raft_python_callback.py write_to_logger 70]  356 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_sync_req_handler, LINE:45
MESSAGE:message pointer and element key/value
VARIABLES:  msg_pointer=x  elem=1 

10/03/2018 22:44:15.0356	[raft_python_callback.py write_to_logger 70]  356 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_sync_req_handler, LINE:45
MESSAGE:message pointer and element key/value
VARIABLES:  msg_pointer=1  elem=(null) 

10/03/2018 22:44:15.0356	[raft_python_callback.py write_to_logger 70]  356 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_sync_req_handler, LINE:55
MESSAGE:leader send sync log msg
VARIABLES:  node->msg_data.sync_res_msg.cmd=add 
	 node->msg_data.sync_res_msg.key=x 
	 node->msg_data.sync_res_msg.value=1 
	

10/03/2018 22:44:15.0356	[raft_python_callback.py write_to_logger 70]  356 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_sync_req_handler, LINE:57
MESSAGE:leader send sync log msg
VARIABLES:  node->msg_data.sync_res_msg.commit_id=0  current_index=0 

10/03/2018 22:44:15.0356	[raft_python_callback.py write_to_logger 70]  356 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_sync_req_handler, LINE:61
MESSAGE:leader sending sync response msg

10/03/2018 22:44:15.0356	[raft_python_callback.py write_to_logger 70]  356 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=180 

10/03/2018 22:44:15.0356	[raft_python_callback.py write_to_logger 70]  356 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=180  fixed_message_size=180 

10/03/2018 22:44:15.0357	[raft_python_callback.py write_to_logger 70]  357 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=180  fixed_message_size=180 

10/03/2018 22:44:15.0357	[raft_python_callback.py write_to_logger 70]  357 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:44:15.0357	[raft_python_callback.py write_to_logger 70]  357 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:44:15.0358	[raft_python_callback.py write_to_logger 70]  358 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_sync_req_handler, LINE:37
MESSAGE:first elemnt - command
VARIABLES:  elem=add 

10/03/2018 22:44:15.0358	[raft_python_callback.py write_to_logger 70]  358 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_sync_req_handler, LINE:45
MESSAGE:message pointer and element key/value
VARIABLES:  msg_pointer=add  elem=xx 

10/03/2018 22:44:15.0358	[raft_python_callback.py write_to_logger 70]  358 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_sync_req_handler, LINE:45
MESSAGE:message pointer and element key/value
VARIABLES:  msg_pointer=xx  elem=1 

10/03/2018 22:44:15.0358	[raft_python_callback.py write_to_logger 70]  358 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_sync_req_handler, LINE:45
MESSAGE:message pointer and element key/value
VARIABLES:  msg_pointer=1  elem=(null) 

10/03/2018 22:44:15.0358	[raft_python_callback.py write_to_logger 70]  358 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_sync_req_handler, LINE:55
MESSAGE:leader send sync log msg
VARIABLES:  node->msg_data.sync_res_msg.cmd=add 
	 node->msg_data.sync_res_msg.key=x 
	 node->msg_data.sync_res_msg.value=1 
	

10/03/2018 22:44:15.0358	[raft_python_callback.py write_to_logger 70]  358 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_sync_req_handler, LINE:57
MESSAGE:leader send sync log msg
VARIABLES:  node->msg_data.sync_res_msg.commit_id=1  current_index=1 

10/03/2018 22:44:15.0358	[raft_python_callback.py write_to_logger 70]  358 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_sync_req_handler, LINE:61
MESSAGE:leader sending sync response msg

10/03/2018 22:44:15.0358	[raft_python_callback.py write_to_logger 70]  358 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=180 

10/03/2018 22:44:15.0358	[raft_python_callback.py write_to_logger 70]  358 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=180  fixed_message_size=180 

10/03/2018 22:44:15.0358	[raft_python_callback.py write_to_logger 70]  358 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=180  fixed_message_size=180 

10/03/2018 22:44:15.0359	[raft_python_callback.py write_to_logger 70]  359 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:44:15.0359	[raft_python_callback.py write_to_logger 70]  359 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:44:16.0185	[raft_python_callback.py write_to_logger 70]  185 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:16.0186	[raft_python_callback.py write_to_logger 70]  186 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:16.0186	[raft_python_callback.py write_to_logger 70]  186 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:44:16.0186	[raft_python_callback.py write_to_logger 70]  186 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:44:16.0186	[raft_python_callback.py write_to_logger 70]  186 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:44:16.0186	[raft_python_callback.py write_to_logger 70]  186 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:44:16.0186	[raft_python_callback.py write_to_logger 70]  186 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:16.0187	[raft_python_callback.py write_to_logger 70]  187 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:16.0187	[raft_python_callback.py write_to_logger 70]  187 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:44:16.0187	[raft_python_callback.py write_to_logger 70]  187 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:44:17.0186	[raft_python_callback.py write_to_logger 70]  186 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:17.0186	[raft_python_callback.py write_to_logger 70]  186 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:17.0186	[raft_python_callback.py write_to_logger 70]  186 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:44:17.0186	[raft_python_callback.py write_to_logger 70]  186 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:44:17.0186	[raft_python_callback.py write_to_logger 70]  186 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:44:17.0186	[raft_python_callback.py write_to_logger 70]  186 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:44:17.0187	[raft_python_callback.py write_to_logger 70]  187 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:17.0187	[raft_python_callback.py write_to_logger 70]  187 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:17.0187	[raft_python_callback.py write_to_logger 70]  187 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:44:17.0187	[raft_python_callback.py write_to_logger 70]  187 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:44:18.0188	[raft_python_callback.py write_to_logger 70]  188 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:18.0188	[raft_python_callback.py write_to_logger 70]  188 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:18.0188	[raft_python_callback.py write_to_logger 70]  188 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:44:18.0188	[raft_python_callback.py write_to_logger 70]  188 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:44:18.0188	[raft_python_callback.py write_to_logger 70]  188 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:44:18.0188	[raft_python_callback.py write_to_logger 70]  188 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:44:18.0189	[raft_python_callback.py write_to_logger 70]  189 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:18.0189	[raft_python_callback.py write_to_logger 70]  189 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:18.0189	[raft_python_callback.py write_to_logger 70]  189 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:44:18.0189	[raft_python_callback.py write_to_logger 70]  189 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:44:19.0188	[raft_python_callback.py write_to_logger 70]  188 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:19.0189	[raft_python_callback.py write_to_logger 70]  189 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:19.0189	[raft_python_callback.py write_to_logger 70]  189 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:44:19.0189	[raft_python_callback.py write_to_logger 70]  189 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:44:19.0189	[raft_python_callback.py write_to_logger 70]  189 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:44:19.0189	[raft_python_callback.py write_to_logger 70]  189 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:44:19.0189	[raft_python_callback.py write_to_logger 70]  189 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:19.0189	[raft_python_callback.py write_to_logger 70]  189 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:19.0189	[raft_python_callback.py write_to_logger 70]  189 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:44:19.0190	[raft_python_callback.py write_to_logger 70]  190 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:44:20.0189	[raft_python_callback.py write_to_logger 70]  189 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:20.0190	[raft_python_callback.py write_to_logger 70]  190 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:20.0190	[raft_python_callback.py write_to_logger 70]  190 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:44:20.0190	[raft_python_callback.py write_to_logger 70]  190 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:44:20.0190	[raft_python_callback.py write_to_logger 70]  190 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:44:20.0190	[raft_python_callback.py write_to_logger 70]  190 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:44:20.0191	[raft_python_callback.py write_to_logger 70]  191 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:20.0191	[raft_python_callback.py write_to_logger 70]  191 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:20.0191	[raft_python_callback.py write_to_logger 70]  191 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:44:20.0191	[raft_python_callback.py write_to_logger 70]  191 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:44:21.0190	[raft_python_callback.py write_to_logger 70]  190 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:21.0190	[raft_python_callback.py write_to_logger 70]  190 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:21.0191	[raft_python_callback.py write_to_logger 70]  191 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:44:21.0191	[raft_python_callback.py write_to_logger 70]  191 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:44:21.0191	[raft_python_callback.py write_to_logger 70]  191 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:44:21.0191	[raft_python_callback.py write_to_logger 70]  191 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:44:21.0191	[raft_python_callback.py write_to_logger 70]  191 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:21.0191	[raft_python_callback.py write_to_logger 70]  191 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:21.0191	[raft_python_callback.py write_to_logger 70]  191 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:44:21.0191	[raft_python_callback.py write_to_logger 70]  191 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:44:22.0008	[raft_python_callback.py write_to_logger 70]  8 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=20  fixed_message_size=20 

10/03/2018 22:44:22.0008	[raft_python_callback.py write_to_logger 70]  8 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:is_relevant_message, LINE:154
MESSAGE:check if msg is relevant
VARIABLES:  sharedRaftData.raft_state.current_state=2  node_message->event=8  is_relevant_rv=1  node_message->term=1 

10/03/2018 22:44:22.0008	[raft_python_callback.py write_to_logger 70]  8 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=1  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:44:22.0008	[raft_python_callback.py write_to_logger 70]  8 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:44:22.0008	[raft_python_callback.py write_to_logger 70]  8 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=8  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:22.0008	[raft_python_callback.py write_to_logger 70]  8 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_vote_req_handler, LINE:254
MESSAGE:leader get vote request msg
VARIABLES:  sharedRaftData.raft_state.term=0 
	 node->term=1 
	 node->message_sent_by=1 
	

10/03/2018 22:44:22.0008	[raft_python_callback.py write_to_logger 70]  8 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:16
MESSAGE:update DB
VARIABLES:  DB_flag=status  key=term 

10/03/2018 22:44:22.0009	[raft_python_callback.py write_to_logger 70]  9 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:17
MESSAGE:update DB
VARIABLES:  value=1 

10/03/2018 22:44:22.0010	[raft_python_callback.py write_to_logger 70]  10 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=7 

10/03/2018 22:44:22.0010	[raft_python_callback.py write_to_logger 70]  10 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_vote_req_handler, LINE:273
MESSAGE:leader sending vote msg because someone else has a bigger term and now he's becoming a follower

10/03/2018 22:44:22.0010	[raft_python_callback.py write_to_logger 70]  10 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=20 

10/03/2018 22:44:22.0010	[raft_python_callback.py write_to_logger 70]  10 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=20  fixed_message_size=20 

10/03/2018 22:44:22.0010	[raft_python_callback.py write_to_logger 70]  10 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:16
MESSAGE:update DB
VARIABLES:  DB_flag=status  key=leader_id 

10/03/2018 22:44:22.0010	[raft_python_callback.py write_to_logger 70]  10 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:17
MESSAGE:update DB
VARIABLES:  value=2 

10/03/2018 22:44:22.0010	[raft_python_callback.py write_to_logger 70]  10 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=20  fixed_message_size=20 

10/03/2018 22:44:22.0011	[raft_python_callback.py write_to_logger 70]  11 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:44:22.0011	[raft_python_callback.py write_to_logger 70]  11 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:44:22.0011	[raft_python_callback.py write_to_logger 70]  11 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:16
MESSAGE:update DB
VARIABLES:  DB_flag=status  key=status 

10/03/2018 22:44:22.0011	[raft_python_callback.py write_to_logger 70]  11 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:17
MESSAGE:update DB
VARIABLES:  value=-3 

10/03/2018 22:44:22.0012	[raft_python_callback.py write_to_logger 70]  12 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:init_raft_queue, LINE:23
MESSAGE:not first time

10/03/2018 22:44:22.0192	[raft_python_callback.py write_to_logger 70]  192 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=0 

10/03/2018 22:44:22.0192	[raft_python_callback.py write_to_logger 70]  192 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=0 

10/03/2018 22:44:25.0542	[raft_python_callback.py write_to_logger 70]  542 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=0 

10/03/2018 22:44:25.0543	[raft_python_callback.py write_to_logger 70]  543 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=0 

10/03/2018 22:44:25.0543	[raft_python_callback.py write_to_logger 70]  543 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:16
MESSAGE:update DB
VARIABLES:  DB_flag=status  key=term 

10/03/2018 22:44:25.0544	[raft_python_callback.py write_to_logger 70]  544 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:17
MESSAGE:update DB
VARIABLES:  value=2 

10/03/2018 22:44:25.0545	[raft_python_callback.py write_to_logger 70]  545 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:16
MESSAGE:update DB
VARIABLES:  DB_flag=status  key=status 

10/03/2018 22:44:25.0545	[raft_python_callback.py write_to_logger 70]  545 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:17
MESSAGE:update DB
VARIABLES:  value=-2 

10/03/2018 22:44:25.0546	[raft_python_callback.py write_to_logger 70]  546 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:16
MESSAGE:update DB
VARIABLES:  DB_flag=status  key=leader_id 

10/03/2018 22:44:25.0546	[raft_python_callback.py write_to_logger 70]  546 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:17
MESSAGE:update DB
VARIABLES:  value=0 

10/03/2018 22:44:25.0547	[raft_python_callback.py write_to_logger 70]  547 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=8 

10/03/2018 22:44:25.0547	[raft_python_callback.py write_to_logger 70]  547 root [DEBUG] 
FILE:./C/Code/follower_handlers.c, FUNCTION:follower_time_out_handler, LINE:253
MESSAGE:follower became candidate and is sending request for vote msg

10/03/2018 22:44:25.0548	[raft_python_callback.py write_to_logger 70]  548 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=20 

10/03/2018 22:44:25.0548	[raft_python_callback.py write_to_logger 70]  548 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=20  fixed_message_size=20 

10/03/2018 22:44:25.0548	[raft_python_callback.py write_to_logger 70]  548 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:44:25.0548	[raft_python_callback.py write_to_logger 70]  548 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:44:25.0548	[raft_python_callback.py write_to_logger 70]  548 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=20  fixed_message_size=20 

10/03/2018 22:44:25.0548	[raft_python_callback.py write_to_logger 70]  548 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:init_raft_queue, LINE:23
MESSAGE:not first time

10/03/2018 22:44:25.0796	[raft_python_callback.py write_to_logger 70]  796 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=20  fixed_message_size=20 

10/03/2018 22:44:25.0796	[raft_python_callback.py write_to_logger 70]  796 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:is_relevant_for_candidate, LINE:77
MESSAGE:candidate check if msg is relevant
VARIABLES:  sharedRaftData.raft_state.term=2  sharedRaftData.raft_state.server_id=2  node_message->message_sent_to=0  node_message->term=2  node_message->event=8 

10/03/2018 22:44:25.0796	[raft_python_callback.py write_to_logger 70]  796 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:is_relevant_for_candidate, LINE:91
MESSAGE:candidate:relevant msg

10/03/2018 22:44:25.0796	[raft_python_callback.py write_to_logger 70]  796 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:is_relevant_message, LINE:154
MESSAGE:check if msg is relevant
VARIABLES:  sharedRaftData.raft_state.current_state=1  node_message->event=8  is_relevant_rv=1  node_message->term=2 

10/03/2018 22:44:25.0796	[raft_python_callback.py write_to_logger 70]  796 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=1  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:44:25.0797	[raft_python_callback.py write_to_logger 70]  797 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:44:25.0796	[raft_python_callback.py write_to_logger 70]  796 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=8  sharedRaftData.raft_state.current_state=1 

10/03/2018 22:44:25.0797	[raft_python_callback.py write_to_logger 70]  797 root [DEBUG] 
FILE:./C/Code/candidate_handlers.c, FUNCTION:candidate_vote_req_handler, LINE:111
MESSAGE:candidate get vote request msg
VARIABLES:  sharedRaftData.raft_state.term=2  sharedRaftData.raft_state.did_I_vote=1  node->term=2  node->message_sent_by=1 

10/03/2018 22:44:28.0891	[raft_python_callback.py write_to_logger 70]  891 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=1 

10/03/2018 22:44:28.0892	[raft_python_callback.py write_to_logger 70]  892 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=1 

10/03/2018 22:44:28.0892	[raft_python_callback.py write_to_logger 70]  892 root [DEBUG] 
FILE:./C/Code/candidate_handlers.c, FUNCTION:candidate_time_out_handler, LINE:152
MESSAGE:timeout event
VARIABLES:  sharedRaftData.raft_state.wakeup_counter=0 

10/03/2018 22:44:28.0892	[raft_python_callback.py write_to_logger 70]  892 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:16
MESSAGE:update DB
VARIABLES:  DB_flag=status  key=term 

10/03/2018 22:44:28.0892	[raft_python_callback.py write_to_logger 70]  892 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:17
MESSAGE:update DB
VARIABLES:  value=3 

10/03/2018 22:44:28.0894	[raft_python_callback.py write_to_logger 70]  894 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=8 

10/03/2018 22:44:28.0894	[raft_python_callback.py write_to_logger 70]  894 root [DEBUG] 
FILE:./C/Code/candidate_handlers.c, FUNCTION:candidate_time_out_handler, LINE:169
MESSAGE:candidate sending request for vote msg
VARIABLES:  sharedRaftData.raft_state.term=3 

10/03/2018 22:44:28.0894	[raft_python_callback.py write_to_logger 70]  894 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=20 

10/03/2018 22:44:28.0894	[raft_python_callback.py write_to_logger 70]  894 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=20  fixed_message_size=20 

10/03/2018 22:44:28.0894	[raft_python_callback.py write_to_logger 70]  894 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=20  fixed_message_size=20 

10/03/2018 22:44:28.0894	[raft_python_callback.py write_to_logger 70]  894 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:44:28.0894	[raft_python_callback.py write_to_logger 70]  894 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:44:32.0241	[raft_python_callback.py write_to_logger 70]  241 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=1 

10/03/2018 22:44:32.0242	[raft_python_callback.py write_to_logger 70]  242 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=1 

10/03/2018 22:44:32.0242	[raft_python_callback.py write_to_logger 70]  242 root [DEBUG] 
FILE:./C/Code/candidate_handlers.c, FUNCTION:candidate_time_out_handler, LINE:152
MESSAGE:timeout event
VARIABLES:  sharedRaftData.raft_state.wakeup_counter=0 

10/03/2018 22:44:32.0247	[raft_python_callback.py write_to_logger 70]  247 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=20  fixed_message_size=20 

10/03/2018 22:44:32.0247	[raft_python_callback.py write_to_logger 70]  247 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:is_relevant_for_candidate, LINE:77
MESSAGE:candidate check if msg is relevant
VARIABLES:  sharedRaftData.raft_state.term=3  sharedRaftData.raft_state.server_id=2  node_message->message_sent_to=0  node_message->term=3  node_message->event=8 

10/03/2018 22:44:32.0247	[raft_python_callback.py write_to_logger 70]  247 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:is_relevant_for_candidate, LINE:91
MESSAGE:candidate:relevant msg

10/03/2018 22:44:32.0247	[raft_python_callback.py write_to_logger 70]  247 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:is_relevant_message, LINE:154
MESSAGE:check if msg is relevant
VARIABLES:  sharedRaftData.raft_state.current_state=1  node_message->event=8  is_relevant_rv=1  node_message->term=3 

10/03/2018 22:44:32.0247	[raft_python_callback.py write_to_logger 70]  247 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=1  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:44:32.0247	[raft_python_callback.py write_to_logger 70]  247 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:44:32.0247	[raft_python_callback.py write_to_logger 70]  247 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=8  sharedRaftData.raft_state.current_state=1 

10/03/2018 22:44:32.0248	[raft_python_callback.py write_to_logger 70]  248 root [DEBUG] 
FILE:./C/Code/candidate_handlers.c, FUNCTION:candidate_vote_req_handler, LINE:111
MESSAGE:candidate get vote request msg
VARIABLES:  sharedRaftData.raft_state.term=3  sharedRaftData.raft_state.did_I_vote=1  node->term=3  node->message_sent_by=1 

10/03/2018 22:44:36.0077	[raft_python_callback.py write_to_logger 70]  77 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=1 

10/03/2018 22:44:36.0078	[raft_python_callback.py write_to_logger 70]  78 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=1 

10/03/2018 22:44:36.0078	[raft_python_callback.py write_to_logger 70]  78 root [DEBUG] 
FILE:./C/Code/candidate_handlers.c, FUNCTION:candidate_time_out_handler, LINE:152
MESSAGE:timeout event
VARIABLES:  sharedRaftData.raft_state.wakeup_counter=1 

10/03/2018 22:44:36.0078	[raft_python_callback.py write_to_logger 70]  78 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:16
MESSAGE:update DB
VARIABLES:  DB_flag=status  key=term 

10/03/2018 22:44:36.0078	[raft_python_callback.py write_to_logger 70]  78 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:17
MESSAGE:update DB
VARIABLES:  value=4 

10/03/2018 22:44:36.0079	[raft_python_callback.py write_to_logger 70]  79 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=8 

10/03/2018 22:44:36.0080	[raft_python_callback.py write_to_logger 70]  80 root [DEBUG] 
FILE:./C/Code/candidate_handlers.c, FUNCTION:candidate_time_out_handler, LINE:169
MESSAGE:candidate sending request for vote msg
VARIABLES:  sharedRaftData.raft_state.term=4 

10/03/2018 22:44:36.0080	[raft_python_callback.py write_to_logger 70]  80 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=20 

10/03/2018 22:44:36.0080	[raft_python_callback.py write_to_logger 70]  80 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=20  fixed_message_size=20 

10/03/2018 22:44:36.0080	[raft_python_callback.py write_to_logger 70]  80 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=20  fixed_message_size=20 

10/03/2018 22:44:36.0080	[raft_python_callback.py write_to_logger 70]  80 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:44:36.0080	[raft_python_callback.py write_to_logger 70]  80 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:44:36.0957	[raft_python_callback.py write_to_logger 70]  957 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=20  fixed_message_size=20 

10/03/2018 22:44:36.0957	[raft_python_callback.py write_to_logger 70]  957 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:is_relevant_for_candidate, LINE:77
MESSAGE:candidate check if msg is relevant
VARIABLES:  sharedRaftData.raft_state.term=4  sharedRaftData.raft_state.server_id=2  node_message->message_sent_to=0  node_message->term=4  node_message->event=8 

10/03/2018 22:44:36.0957	[raft_python_callback.py write_to_logger 70]  957 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:is_relevant_for_candidate, LINE:91
MESSAGE:candidate:relevant msg

10/03/2018 22:44:36.0957	[raft_python_callback.py write_to_logger 70]  957 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:is_relevant_message, LINE:154
MESSAGE:check if msg is relevant
VARIABLES:  sharedRaftData.raft_state.current_state=1  node_message->event=8  is_relevant_rv=1  node_message->term=4 

10/03/2018 22:44:36.0958	[raft_python_callback.py write_to_logger 70]  958 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=1  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:44:36.0958	[raft_python_callback.py write_to_logger 70]  958 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:44:36.0958	[raft_python_callback.py write_to_logger 70]  958 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=8  sharedRaftData.raft_state.current_state=1 

10/03/2018 22:44:36.0958	[raft_python_callback.py write_to_logger 70]  958 root [DEBUG] 
FILE:./C/Code/candidate_handlers.c, FUNCTION:candidate_vote_req_handler, LINE:111
MESSAGE:candidate get vote request msg
VARIABLES:  sharedRaftData.raft_state.term=4  sharedRaftData.raft_state.did_I_vote=1  node->term=4  node->message_sent_by=1 

10/03/2018 22:44:39.0913	[raft_python_callback.py write_to_logger 70]  913 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=1 

10/03/2018 22:44:39.0914	[raft_python_callback.py write_to_logger 70]  914 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=1 

10/03/2018 22:44:39.0914	[raft_python_callback.py write_to_logger 70]  914 root [DEBUG] 
FILE:./C/Code/candidate_handlers.c, FUNCTION:candidate_time_out_handler, LINE:152
MESSAGE:timeout event
VARIABLES:  sharedRaftData.raft_state.wakeup_counter=0 

10/03/2018 22:44:39.0914	[raft_python_callback.py write_to_logger 70]  914 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:16
MESSAGE:update DB
VARIABLES:  DB_flag=status  key=term 

10/03/2018 22:44:39.0914	[raft_python_callback.py write_to_logger 70]  914 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:17
MESSAGE:update DB
VARIABLES:  value=5 

10/03/2018 22:44:39.0915	[raft_python_callback.py write_to_logger 70]  915 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=8 

10/03/2018 22:44:39.0915	[raft_python_callback.py write_to_logger 70]  915 root [DEBUG] 
FILE:./C/Code/candidate_handlers.c, FUNCTION:candidate_time_out_handler, LINE:169
MESSAGE:candidate sending request for vote msg
VARIABLES:  sharedRaftData.raft_state.term=5 

10/03/2018 22:44:39.0916	[raft_python_callback.py write_to_logger 70]  916 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=20 

10/03/2018 22:44:39.0916	[raft_python_callback.py write_to_logger 70]  916 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=20  fixed_message_size=20 

10/03/2018 22:44:39.0916	[raft_python_callback.py write_to_logger 70]  916 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=20  fixed_message_size=20 

10/03/2018 22:44:39.0916	[raft_python_callback.py write_to_logger 70]  916 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:44:39.0916	[raft_python_callback.py write_to_logger 70]  916 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:44:42.0588	[raft_python_callback.py write_to_logger 70]  588 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=20  fixed_message_size=20 

10/03/2018 22:44:42.0588	[raft_python_callback.py write_to_logger 70]  588 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:is_relevant_for_candidate, LINE:77
MESSAGE:candidate check if msg is relevant
VARIABLES:  sharedRaftData.raft_state.term=5  sharedRaftData.raft_state.server_id=2  node_message->message_sent_to=0  node_message->term=5  node_message->event=8 

10/03/2018 22:44:42.0589	[raft_python_callback.py write_to_logger 70]  589 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:is_relevant_for_candidate, LINE:91
MESSAGE:candidate:relevant msg

10/03/2018 22:44:42.0589	[raft_python_callback.py write_to_logger 70]  589 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:is_relevant_message, LINE:154
MESSAGE:check if msg is relevant
VARIABLES:  sharedRaftData.raft_state.current_state=1  node_message->event=8  is_relevant_rv=1  node_message->term=5 

10/03/2018 22:44:42.0589	[raft_python_callback.py write_to_logger 70]  589 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=1  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:44:42.0589	[raft_python_callback.py write_to_logger 70]  589 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:44:42.0589	[raft_python_callback.py write_to_logger 70]  589 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=8  sharedRaftData.raft_state.current_state=1 

10/03/2018 22:44:42.0589	[raft_python_callback.py write_to_logger 70]  589 root [DEBUG] 
FILE:./C/Code/candidate_handlers.c, FUNCTION:candidate_vote_req_handler, LINE:111
MESSAGE:candidate get vote request msg
VARIABLES:  sharedRaftData.raft_state.term=5  sharedRaftData.raft_state.did_I_vote=1  node->term=5  node->message_sent_by=1 

10/03/2018 22:44:42.0837	[raft_python_callback.py write_to_logger 70]  837 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=1 

10/03/2018 22:44:42.0837	[raft_python_callback.py write_to_logger 70]  837 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=1 

10/03/2018 22:44:42.0837	[raft_python_callback.py write_to_logger 70]  837 root [DEBUG] 
FILE:./C/Code/candidate_handlers.c, FUNCTION:candidate_time_out_handler, LINE:152
MESSAGE:timeout event
VARIABLES:  sharedRaftData.raft_state.wakeup_counter=0 

10/03/2018 22:44:42.0837	[raft_python_callback.py write_to_logger 70]  837 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:16
MESSAGE:update DB
VARIABLES:  DB_flag=status  key=term 

10/03/2018 22:44:42.0838	[raft_python_callback.py write_to_logger 70]  838 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:17
MESSAGE:update DB
VARIABLES:  value=6 

10/03/2018 22:44:42.0839	[raft_python_callback.py write_to_logger 70]  839 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=8 

10/03/2018 22:44:42.0839	[raft_python_callback.py write_to_logger 70]  839 root [DEBUG] 
FILE:./C/Code/candidate_handlers.c, FUNCTION:candidate_time_out_handler, LINE:169
MESSAGE:candidate sending request for vote msg
VARIABLES:  sharedRaftData.raft_state.term=6 

10/03/2018 22:44:42.0839	[raft_python_callback.py write_to_logger 70]  839 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=20 

10/03/2018 22:44:42.0839	[raft_python_callback.py write_to_logger 70]  839 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=20  fixed_message_size=20 

10/03/2018 22:44:42.0839	[raft_python_callback.py write_to_logger 70]  839 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:44:42.0839	[raft_python_callback.py write_to_logger 70]  839 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:44:42.0840	[raft_python_callback.py write_to_logger 70]  840 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=20  fixed_message_size=20 

10/03/2018 22:44:42.0998	[raft_python_callback.py write_to_logger 70]  998 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=20  fixed_message_size=20 

10/03/2018 22:44:42.0998	[raft_python_callback.py write_to_logger 70]  998 root [INFO ] 
FILE:./C/Code/handlers_utils.c, FUNCTION:check_if_is_for_me, LINE:103
MESSAGE:got new message for me

10/03/2018 22:44:42.0998	[raft_python_callback.py write_to_logger 70]  998 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:is_relevant_for_candidate, LINE:77
MESSAGE:candidate check if msg is relevant
VARIABLES:  sharedRaftData.raft_state.term=6  sharedRaftData.raft_state.server_id=2  node_message->message_sent_to=2  node_message->term=6  node_message->event=7 

10/03/2018 22:44:42.0998	[raft_python_callback.py write_to_logger 70]  998 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:is_relevant_for_candidate, LINE:91
MESSAGE:candidate:relevant msg

10/03/2018 22:44:42.0998	[raft_python_callback.py write_to_logger 70]  998 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:is_relevant_message, LINE:154
MESSAGE:check if msg is relevant
VARIABLES:  sharedRaftData.raft_state.current_state=1  node_message->event=7  is_relevant_rv=1  node_message->term=6 

10/03/2018 22:44:42.0998	[raft_python_callback.py write_to_logger 70]  998 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=1  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:44:42.0998	[raft_python_callback.py write_to_logger 70]  998 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:44:42.0999	[raft_python_callback.py write_to_logger 70]  999 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=7  sharedRaftData.raft_state.current_state=1 

10/03/2018 22:44:42.0999	[raft_python_callback.py write_to_logger 70]  999 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:init_raft_queue, LINE:23
MESSAGE:not first time

10/03/2018 22:44:42.0999	[raft_python_callback.py write_to_logger 70]  999 root [DEBUG] 
FILE:./C/Code/candidate_handlers.c, FUNCTION:candidate_vote_for_me_handler, LINE:20
MESSAGE:candidate got majority- becoming leader
VARIABLES:  node->event=7 
	 sharedRaftData.raft_state.term=6 
	 sharedRaftData.raft_state.current_state=2 
	 sharedRaftData.raft_state.vote_counter=2 
	 node->message_sent_by=1 
	 node->term=6 
	

10/03/2018 22:44:42.0999	[raft_python_callback.py write_to_logger 70]  999 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:16
MESSAGE:update DB
VARIABLES:  DB_flag=status  key=status 

10/03/2018 22:44:42.0999	[raft_python_callback.py write_to_logger 70]  999 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:17
MESSAGE:update DB
VARIABLES:  value=-1 

10/03/2018 22:44:43.0000	[raft_python_callback.py write_to_logger 70]  0 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:16
MESSAGE:update DB
VARIABLES:  DB_flag=status  key=leader_id 

10/03/2018 22:44:43.0000	[raft_python_callback.py write_to_logger 70]  0 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:17
MESSAGE:update DB
VARIABLES:  value=2 

10/03/2018 22:44:43.0002	[raft_python_callback.py write_to_logger 70]  2 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:44:43.0002	[raft_python_callback.py write_to_logger 70]  2 root [DEBUG] 
FILE:./C/Code/candidate_handlers.c, FUNCTION:candidate_vote_for_me_handler, LINE:40
MESSAGE:candidate sending keep alive hb msg

10/03/2018 22:44:43.0002	[raft_python_callback.py write_to_logger 70]  2 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:44:43.0002	[raft_python_callback.py write_to_logger 70]  2 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:43.0002	[raft_python_callback.py write_to_logger 70]  2 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:44:43.0002	[raft_python_callback.py write_to_logger 70]  2 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:44:43.0002	[raft_python_callback.py write_to_logger 70]  2 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:43.0002	[raft_python_callback.py write_to_logger 70]  2 root [DEBUG] 
FILE:./C/Code/candidate_handlers.c, FUNCTION:candidate_vote_for_me_handler, LINE:68
MESSAGE:end of vote event
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:46.0539	[raft_python_callback.py write_to_logger 70]  539 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:46.0539	[raft_python_callback.py write_to_logger 70]  539 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:46.0539	[raft_python_callback.py write_to_logger 70]  539 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:44:46.0539	[raft_python_callback.py write_to_logger 70]  539 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:44:46.0539	[raft_python_callback.py write_to_logger 70]  539 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:44:46.0539	[raft_python_callback.py write_to_logger 70]  539 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:44:46.0540	[raft_python_callback.py write_to_logger 70]  540 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:46.0540	[raft_python_callback.py write_to_logger 70]  540 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:44:46.0540	[raft_python_callback.py write_to_logger 70]  540 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

10/03/2018 22:44:46.0540	[raft_python_callback.py write_to_logger 70]  540 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:47.0539	[raft_python_callback.py write_to_logger 70]  539 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:49
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:47.0539	[raft_python_callback.py write_to_logger 70]  539 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=2 

10/03/2018 22:44:47.0539	[raft_python_callback.py write_to_logger 70]  539 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:129
MESSAGE:leader timeout event

10/03/2018 22:44:47.0540	[raft_python_callback.py write_to_logger 70]  540 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=1 

10/03/2018 22:44:47.0540	[raft_python_callback.py write_to_logger 70]  540 root [DEBUG] 
FILE:./C/Code/leader_handlers.c, FUNCTION:leader_time_out_handler, LINE:133
MESSAGE:leader sending keep alive hb msg

10/03/2018 22:44:47.0540	[raft_python_callback.py write_to_logger 70]  540 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=24 

10/03/2018 22:44:47.0540	[raft_python_callback.py write_to_logger 70]  540 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:120
MESSAGE:msg was successfully sent
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:47.0540	[raft_python_callback.py write_to_logger 70]  540 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:156
MESSAGE:msg was successfully read
VARIABLES:  message_size=24  fixed_message_size=24 

10/03/2018 22:44:47.0540	[raft_python_callback.py write_to_logger 70]  540 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:20
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=2  sharedRaftData.raft_state.server_id=2 

10/03/2018 22:44:47.0540	[raft_python_callback.py write_to_logger 70]  540 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:133
MESSAGE:try to read message

